src/justfile.rs

```diff
@@ -28,19 +28,31 @@ impl<'src> Justfile<'src> {
     self.recipes.len()
   }
 
-  pub(crate) fn suggest(&self, name: &str) -> Option<&'src str> {
+  pub(crate) fn suggest(&self, name: &str) -> Option<Suggestion> {
     let mut suggestions = self
       .recipes
       .keys()
-      .map(|suggestion| (edit_distance(suggestion, name), suggestion))
-      .collect::<Vec<_>>();
-    suggestions.sort();
-    if let Some(&(distance, suggestion)) = suggestions.first() {
-      if distance < 3 {
-        return Some(suggestion);
-      }
+      .map(|recipe_name| {
+        (edit_distance(recipe_name, name), Suggestion {
+          name:   recipe_name,
+          target: None,
+        })
+      })
+      .chain(self.aliases.iter().map(|(alias_name, alias)| {
+        (edit_distance(alias_name, name), Suggestion {
+          name:   alias_name,
+          target: Some(alias.target.name.lexeme()),
+        })
+      }))
+      .filter(|(distance, _suggestion)| distance < &3)
+      .collect::<Vec<(usize, Suggestion)>>();
+    suggestions.sort_by_key(|(distance, _suggestion)| *distance);
+
+    if let Some((_distance, suggestion)) = suggestions.first() {
```

Comment:
This is totally fine, but you could also do:

```
suggestions.into_iter().map(|(_distance, suggestion)| suggestion).next()
```

I think that `into_iter().next()` will give you the value, and not a reference to the value, so you don't need to derference with `*`.

Reply:
Oh good point. I'd thought of doing the map before, but I was worried about needlessly applying the function to the entire list when I just needed the first element. I'd forgot that Rust uses lazy processing there, so it's not a problem.

Reply:
It's still wild to me that many uses of closures in rust are as efficient as they are. In my head I still think of things like closures, map, etc, as being spooky, inefficient things that you can only use if you don't care about performance.

Reply:
Haha definitely! I've been using JS too much.